rest-service/manager_rest/upload_manager.py

```diff
@@ -257,7 +258,16 @@ def _move_archive_to_uploaded_dir(self,
                                 .format(uploaded_dir))
         if os.path.isfile(archive_path):
             if not dest_file_name:
-                archive_type = self._get_archive_type(archive_path)
+                try:
+                    archive_type = self._get_archive_type(archive_path)
+                except RuntimeError as e:
+                    if 'Can\'t recognize archive type' in str(e):
```

Comment:
it's not nice, checking errors by their human-readable text is super fragile - we raise that exception, so make it a more specific error rather than RuntimeError, and `except` on that (you can make the new error a subclass of RuntimeError for back-compat)